lvsfunc/source.py

```diff
@@ -0,0 +1,206 @@
+                 vps_indexer: Callable[..., vs.VideoNode] = core.dgdecodenv.DGSource, ext: str = '.dgi') -> None:
+        super().__init__(path, vps_indexer, ext)
+
+    def get_cmd(self, files: List[Path], output: Path) -> List[Any]:
+        return [self.path, '-i', ','.join(map(str, files)), '-o', output, '-h']
+
+
+def dvd_source(vob_folder: Union[Path, str], idx: DVDIndexer = D2VWitch(), ifo_file: Optional[Union[Path, str]] = None,
+               extra: bool = False, **kwargs: Any) -> Tuple[List[vs.VideoNode], List[List[int]]]:
+    """Concat .vob DVD files and trim them per titles using the .ifo file as reference.
+
+    :param vob_folder:      Folder with .vob files.
+    :param idx:             DVD indexer to be used.
+    :param ifo_file:        Optional IFO file path. If not specified, will search 'VTS_01_0.IFO'.
+    :param extra:           Adds the splash screen and BD Menu to the return clips.
+    :kwargs:                Arguments passed to the indexing filter.
+    :return:                Returns a tuple of trimmed clips per title and the chapter frame numbers.
+    """
+    vob_folder = Path(vob_folder)
+
+    # Index vob files using idx
+    vob_files = sorted(vob_folder.glob('*.vob'))
+
+    if not (output := Path(vob_files[0].with_suffix(idx.ext))).exists():
+        run(idx.get_cmd(vob_files, output), check=True, text=True, encoding='utf-8')
+
+    all_titles = idx.vps_indexer(str(output), **kwargs)
+
+    # Parse IFO info
+    if not ifo_file:
+        ifo_file = vob_folder / 'VTS_01_0.IFO'
+    with open(ifo_file, 'rb') as file:
+        pgci = vts_ifo.load_vts_pgci(cast(BufferedReader, file))
+
+    chapters_frames: List[List[int]] = []
+    for prog in pgci.program_chains:
+        dvd_fps_s = [pb_time.fps for pb_time in prog.playback_times]
+        if all(dvd_fps_s[0] == dvd_fps for dvd_fps in dvd_fps_s):
+            fps = vts_ifo.FRAMERATE[dvd_fps_s[0]]
+        else:
+            raise ValueError('dvd_source: No VFR allowed!')
+
+        raw_fps = 30 if fps.numerator == 30000 else 25
+
+        # Convert in frames
+        chapters_frames.append([0] + [
+            pb_time.frames
+            + (pb_time.hours * 3600 + pb_time.minutes * 60 + pb_time.seconds) * raw_fps
+            for pb_time in prog.playback_times]
+        )
+
+    chapters_frames = [
+        list(accumulate(chapter_frames))
+        for chapter_frames in chapters_frames
+    ]
+
+    durations = list(accumulate([0] + [frame[-1] for frame in chapters_frames]))
+
+    # Remove splash screen and DVD Menu
+    clip = all_titles[-durations[-1]:]
+
+    # Trim per title
+    clips = [clip[s:e] for s, e in zip(durations[:-1], durations[1:])]
+
+    if extra:
+        clips.insert(0, all_titles[:-durations[-1]])
+
+    return clips, chapters_frames
+
+
+def trim_clips_per_chapters(clips: List[vs.VideoNode], chapters_frames: List[List[int]]) -> List[List[vs.VideoNode]]:
+    """Trim the ouputs of `dvd_source` per chapters.
```

Comment:
"per chapters" -> "by chapter". Same for the `:return:`.

Reply:
Fixed in https://github.com/Irrational-Encoding-Wizardry/lvsfunc/pull/59/commits/82ed70d7deb1c4d909fe08bdca9a7b093f83aa0d